### PR TITLE
Add a section on how to extend addons

### LINKED ISSUE #170: Overriding files from addons
It would be useful to document how to override files from addons. For example, if an addon contains `services/foo.js`, and an app wants to extend that service to add their own behavior, how would they do that?

### PR DESCRIPTION
**Pull request by mehulkar**
Closes #170

### CODE REVIEW

File: guides/basic-use/using-addons.md [hunk 1]
```diff
@@ -48,6 +48,53 @@ Here are examples of popular community-maintained addons that are unique to Embe
...
+- `addon/services/state-manager.js`, and
+- `app/services/state-manager.js`
+
+Each of these files has a special purpose. The first file (in the `addon/` directory), contains
+the source code of the `StateManager` class  and is *importable* with ES6 imports. The second file
+contains only an import and an export statement, making it available for the *resolver*.
+In order to extend this service, we must use both these key concepts.
+
+In other words, we must export an extended version of the *importable* code at the *resolvable* path.
+More conceretely, we must create a fie at `app/services/state-manager.js`  that imports the StateManager
```
Comment (rajasegar) [inline guides/basic-use/using-addons.md:70]:
```suggestion
More concretely, we must create a file at `app/services/state-manager.js`  that imports the StateManager
```

File: guides/basic-use/using-addons.md [hunk 2]
```diff
@@ -48,6 +48,53 @@ Here are examples of popular community-maintained addons that are unique to Embe
...
 There are too many amazing addons to mention them all here! We thank everyone who shares their work with others through public addons.
 
+### Extending Addons
+
+Sometimes, it can be useful to extend code provided by an addon. Because runtime
+code in an Ember app is *resolved*, rather than explicitly imported, the mechanism to overwrite it
+is by placing a file where Ember's resolver expects it in the directory structure. Let's take a look
+at how to do this through an example.
+
+Consider an addon `ember-state-manager` that contains a `Service` class called called `StateManager`.
```
Comment (rajasegar) [inline guides/basic-use/using-addons.md:58]:
```suggestion
Consider an addon `ember-state-manager` that contains a `Service` class called `StateManager`.
```

File: guides/basic-use/using-addons.md [hunk 3]
```diff
@@ -48,6 +48,53 @@ Here are examples of popular community-maintained addons that are unique to Embe
...
+- `app/services/state-manager.js`
+
+Each of these files has a special purpose. The first file (in the `addon/` directory), contains
+the source code of the `StateManager` class  and is *importable* with ES6 imports. The second file
+contains only an import and an export statement, making it available for the *resolver*.
+In order to extend this service, we must use both these key concepts.
+
+In other words, we must export an extended version of the *importable* code at the *resolvable* path.
+More conceretely, we must create a fie at `app/services/state-manager.js`  that imports the StateManager
+class and extends it using Javascript classes:
```
Comment (rajasegar) [inline guides/basic-use/using-addons.md:71]:
```suggestion
class and extends it using JavaScript classes:
```

**Comment by rajasegar** [comment]
@mehulkar Also you might want to add `StateManager` here
https://github.com/ember-learn/cli-guides/blob/220d733d84cade45a9831e3a394ef9265f0f6747/.local.dic#L86
to fix lint errors

File: guides/basic-use/using-addons.md [hunk 4]
```diff
@@ -48,6 +48,53 @@ Here are examples of popular community-maintained addons that are unique to Embe
...
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+
+export default Button extends Component {
+  // ExtendedStateManager!
+  @service stateManager;
+}
+```
+
+Although this example is for a Service class, the same basic mechanism can be used for anything else
```
Comment (Gaurav0) [inline guides/basic-use/using-addons.md:95]:
```suggestion
Although this example is for a Service class, the same idea can be used for anything else
```

**Comment by mehulkar** [comment]
@rajasegar I think I was just missing some backticks. Should pass now?

**Comment by mehulkar** [comment]
cc @jenweber @locks for review!

**Pull request merged**